src/Microsoft.OData.Client/DataServiceClientResponsePipelineConfiguration.cs

```diff
@@ -309,6 +309,41 @@ internal void ExecuteOnNavigationEndActions(ODataNestedResourceInfo link)
             }
         }
 
+        /// <summary>
+        /// Executes the OData delta resource set start actions.
+        /// </summary>
+        /// <param name="feed">The feed.</param>
+        internal void ExecuteOnDeltaFeedStartActions(ODataDeltaResourceSet deltaFeed)
+        {
+            Debug.Assert(deltaFeed != null, "deltaFeed != null");
+            if (this.readingStartFeedActions.Count > 0)
+            {
+                ReadingDeltaResourceSetArgs args = new ReadingDeltaResourceSetArgs(deltaFeed);
+                foreach (Action<ReadingDeltaResourceSetArgs> feedAction in this.readingStartFeedActions)

```

Comment:
I believe `readingStartFeedActions` and `readingEndFeedActions` preceded the `deltaFeed` feature. There are defined as follows:
```
private readonly List<Action<ReadingFeedArgs>> readingStartFeedActions;
private readonly List<Action<ReadingFeedArgs>> readingEndFeedActions;
```
How are we repurposing them to work with `ReadDeltaResourceSetArgs`?

Reply:
yes

Reply:
I'm thinking that somewhere we should have:
```csharp
private readonly List<Action<ReadingDeltaResourceSetArgs>> readingStartDeltaFeedActions; private readonly List<Action<ReadingDeltaResourceSetArgs>> readingEndDeltaFeedActions;
```
Those collection of delegates are what we'd then reference here.

If my thinking is correct, then the conclusion would be that there's no test or tests that currently validate this logic here...

Reply:
> yes

How, yet `ReadingDeltaResourceSetArgs` don't derive from `ReadingFeedArgs`?

Reply:
ok. I see. fixing this and adding tests